Source/WebKit/UIProcess/WebBackForwardCache.cpp

```diff
@@ -51,23 +51,29 @@ WebBackForwardCache::~WebBackForwardCache()
     clear();
 }
 
+Ref<WebProcessPool> WebBackForwardCache::protectedProcessPool() const
+{
+    return m_processPool.get();
+}
+
 inline void WebBackForwardCache::removeOldestEntry()
 {
     ASSERT(!m_itemsWithCachedPage.isEmptyIgnoringNullReferences());
     if (RefPtr item = m_itemsWithCachedPage.tryTakeFirst())
         item->setBackForwardCacheEntry(nullptr);
 }
 
-void WebBackForwardCache::setCapacity(unsigned capacity)
+void WebBackForwardCache::setCapacity(WebProcessPool& pool, unsigned capacity)
```

Comment:
That's a bit of a weird one, I think we should just use `m_processPool` and protect it. It doesn't make much sense to pass a parameter in that we already have. This is a bit of an error prone API. Seems we're trying to optimize an extra Ref out but it doesn't seem like hot code.

Reply:
But that doesn't work because this function gets called during inside WebProcessPool's constructor:
```
Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   CoreFoundation                	       0x187aff4e8 CFRetain.cold.1 + 16
1   CoreFoundation                	       0x187918958 CFRetain + 140
2   WebKit                        	       0x301510a94 API::Object::ref() const + 32 (APIObject.mm:131)
3   WebKit                        	       0x3023dcd4c WTF::DefaultRefDerefTraits<WebKit::WebProcessPool>::ref(WebKit::WebProcessPool&) + 28 (Ref.h:55)
4   WebKit                        	       0x3023dcd18 WTF::Ref<WebKit::WebProcessPool, WTF::RawPtrTraits<WebKit::WebProcessPool>, WTF::DefaultRefDerefTraits<WebKit::WebProcessPool>>::Ref(WebKit::WebProcessPool&) + 40 (Ref.h:86)
5   WebKit                        	       0x3023af788 WTF::Ref<WebKit::WebProcessPool, WTF::RawPtrTraits<WebKit::WebProcessPool>, WTF::DefaultRefDerefTraits<WebKit::WebProcessPool>>::Ref(WebKit::WebProcessPool&) + 40 (Ref.h:87)
6   WebKit                        	       0x3023af678 WebKit::WebBackForwardCache::protectedProcessPool() const + 52 (WebBackForwardCache.cpp:56)
7   WebKit                        	       0x3023af81c WebKit::WebBackForwardCache::setCapacity(unsigned int) + 132 (WebBackForwardCache.cpp:75)
8   WebKit                        	       0x302720ba8 WebKit::WebProcessPool::updateBackForwardCacheCapacity() + 164 (WebProcessPool.cpp:1506)
9   WebKit                        	       0x30271fba8 WebKit::WebProcessPool::WebProcessPool(API::ProcessPoolConfiguration&) + 2288 (WebProcessPool.cpp:306)
10  WebKit                        	       0x30271f0d4 WebKit::WebProcessPool::WebProcessPool(API::ProcessPoolConfiguration&) + 40 (WebProcessPool.cpp:252)
11  WebKit                        	       0x301946864 void API::Object::constructInWrapper<WebKit::WebProcessPool, API::ProcessPoolConfiguration&>(id<WKObject>, API::ProcessPoolConfiguration&) + 64 (APIObject.h:243)
12  WebKit                        	       0x301946808 -[WKProcessPool _initWithConfiguration:] + 140 (WKProcessPool.mm:102)
```

Reply:
That's unfortunate but I don't think we should make the code look worse to silence a warning. We could silence the warning or call `updateBackForwardCacheCapacity()` from `WebProcessPool::create()` instead of the constructor

Reply:
Another alternative is to drop `m_processPool` and pass the processPool as parameter like you did. It seems `m_processPool` is only used by `setCapacity()`.

Reply:
Yeah, maybe dropping m_processPool is the best approach. No need to keep a reference around if we can just pass in at call site.